mina-transaction/v2/leaves.ts

```diff
@@ -0,0 +1,338 @@
+// NOTE: these leaves are currently backwards compatible with the old encoding format, but the
```

Comment:
there's a few TODOs and nice to haves in here, I propose we get back to them in the future given we target an incremental approach with lots of iterations